shared/src/api/start_game.rs

```diff
@@ -0,0 +1,60 @@
+use serde::{Deserialize, Serialize};
+
+////////////////////////////////////////////////////////////////
+// Request //
+////////////////////////////////////////////////////////////////
+
+#[derive(Serialize, Deserialize)]
+pub struct Request {
+    game_name: String,
+    player_name: String,
+}
```

Comment:
this is basically `shared/src/game_request`, but I realized two things:

- We need a response type too, and since its a response to a specific request maybe they should be bundled (so I renamed it to `start_game` as in the `start_game::Request` and `start_game::Response`.)
- All of these are under the category of api requests, and maybe it would be good to differentiate api stuff from other shared stuff (so I put them in an `api/` directory).